e2e/evm/test.go

```diff
@@ -34,83 +31,46 @@ type TestClient interface {
-	client             TestClient
-	client2            TestClient
-	gasPricer          calls.GasPricer
-	bridgeAddr         common.Address
-	bridgeAddr2        common.Address
-	erc20HandlerAddr   common.Address
-	erc20ContractAddr  common.Address
-	erc721HandlerAddr  common.Address
-	erc721ContractAddr common.Address
-	genericHandlerAddr common.Address
-	assetStoreAddr     common.Address
-	fabric1            calls.TxFabric
-	fabric2            calls.TxFabric
-	endpoint1          string
-	endpoint2          string
-	adminKey           *secp256k1.Keypair
-	erc20RID           [32]byte
-	erc721RID          [32]byte
-	genericRID         [32]byte
+	client1    TestClient
+	client2    TestClient
+	gasPricer  calls.GasPricer
+	fabric1    calls.TxFabric
+	fabric2    calls.TxFabric
+	erc20RID   [32]byte
+	erc721RID  [32]byte
+	genericRID [32]byte
+	config1    local.EVME2EConfig
+	config2    local.EVME2EConfig
 }
 
 func (s *IntegrationTestSuite) SetupSuite() {
-	ethClient, err := evmclient.NewEVMClientFromParams(s.endpoint1, s.adminKey.PrivateKey())
+	config1, err := local.PrepareLocalEVME2EEnv(s.client1, s.fabric1, 1, big.NewInt(2), s.client1.From())
 	if err != nil {
 		panic(err)
 	}
-	s.client = ethClient
+	s.config1 = config1
 
-	ethClient2, err := evmclient.NewEVMClientFromParams(s.endpoint2, s.adminKey.PrivateKey())
+	config2, err := local.PrepareLocalEVME2EEnv(s.client2, s.fabric2, 2, big.NewInt(2), s.client2.From())
 	if err != nil {
 		panic(err)
 	}
-	s.client2 = ethClient2
+	s.config2 = config2
 
-	b, err := ethClient.LatestBlock()
-	if err != nil {
-		panic(err)
-	}
-
-	log.Debug().Msgf("Latest block %s", b.String())
-
-	config, err := local.PrepareLocalEVME2EEnv(ethClient, s.fabric1, 1, big.NewInt(2), s.adminKey.CommonAddress())
-	if err != nil {
-		panic(err)
-	}
-
-	s.bridgeAddr = config.BridgeAddr
-	s.erc20ContractAddr = config.Erc20Addr
-	s.erc20HandlerAddr = config.Erc20HandlerAddr
-	s.erc721ContractAddr = config.Erc721Addr
-	s.erc721HandlerAddr = config.Erc721HandlerAddr
-	s.genericHandlerAddr = config.GenericHandlerAddr
-	s.assetStoreAddr = config.AssetStoreAddr
-
-	s.gasPricer = evmgaspricer.NewStaticGasPriceDeterminant(s.client, nil)
-
-	cfg2, err := local.PrepareLocalEVME2EEnv(ethClient2, s.fabric2, 2, big.NewInt(2), s.adminKey.CommonAddress())
-	if err != nil {
-		panic(err)
-	}
-
-	s.bridgeAddr2 = cfg2.BridgeAddr
-	s.erc20RID = calls.SliceTo32Bytes(append(common.LeftPadBytes(config.Erc20Addr.Bytes(), 31), uint8(0)))
-	s.genericRID = calls.SliceTo32Bytes(append(common.LeftPadBytes(config.GenericHandlerAddr.Bytes(), 31), uint8(1)))
-	s.erc721RID = calls.SliceTo32Bytes(append(common.LeftPadBytes(config.Erc721Addr.Bytes(), 31), uint8(2)))
+	s.erc20RID = calls.SliceTo32Bytes(common.LeftPadBytes([]byte{0}, 31))
+	s.genericRID = calls.SliceTo32Bytes(common.LeftPadBytes([]byte{1}, 31))
+	s.erc721RID = calls.SliceTo32Bytes(common.LeftPadBytes([]byte{2}, 31))
+	s.gasPricer = evmgaspricer.NewStaticGasPriceDeterminant(s.client2, nil)
```

Comment:
Should we here use `s.client1`?

Reply:
Yeah, I just changed to have s.gasPricer1 and s.gasPricer2 as my tests on Optimism failed :smile: